tests/components/roborock/test_init.py

```diff
@@ -38,3 +40,69 @@ async def test_config_entry_not_ready(
     ):
         await async_setup_component(hass, DOMAIN, {})
         assert mock_roborock_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_config_entry_not_ready_home_data(
+    hass: HomeAssistant, mock_roborock_entry: MockConfigEntry
+) -> None:
+    """Test that when we fail to get home data, entry retries."""
+    with patch(
+        "homeassistant.components.roborock.RoborockApiClient.get_home_data",
+        side_effect=RoborockException(),
+    ), patch(
+        "homeassistant.components.roborock.RoborockDataUpdateCoordinator._async_update_data",
```

Comment:
I think the preferred approach is to patch the client library, and not home assistant internals so that refactoring doesn't require test updates. Can this raise a roborock client exception instead?

Reply:
tbh idk why I didn't to begin with. Fixed now.